src/libPMacc/include/particles/memory/dataTypes/Particle.hpp

```diff
@@ -139,19 +151,29 @@ struct Particle : public InheritLinearly<typename T_FrameType::MethodsList>
 namespace traits
 {
 
-template<typename T_Key,
-typename T_FrameType
+template<
+typename T_Key,
+typename T_FrameType,
+typename T_ValueTypeSeq
 >
 struct HasIdentifier<
-PMacc::Particle<T_FrameType>,
+PMacc::Particle< T_FrameType, T_ValueTypeSeq >,
 T_Key
 >
 {
 private:
-    typedef T_FrameType FrameType;
+    typedef PMacc::Particle<T_FrameType, T_ValueTypeSeq> ParticleType;
+    typedef typename ParticleType::ValueTypeSeq ValueTypeSeq;
 public:
-    typedef typename HasIdentifier<FrameType, T_Key>::type type;
-    BOOST_STATIC_CONSTEXPR bool value = type::value;
+    /* if SolvedAliasName is void_ than we have no T_Key in our Sequence.
```

Comment:
If T_Key can not be found in the T_ValueTypeSeq of this Particle class, SolvedAliasName will we void_.